Automattic/delphin
# Add `wpcomLoginMiddleware` for logging the user into WordPress.com

This PR adds `wpcom-login-middleware`, client middleware which logs the user into WordPress.com when the `WPCOM_LOG_IN` event is dispatched. We will eventually use this to redirect the user to a get.blog-specific signup flow in Calypso, but currently it just logs the user in to WordPress.com and redirects them to `https://wordpress.com`. The log in currently opens in a new window.

This PR also logs the user in to their WordPress.com account in the background after they enter their verification code.

**Note:** This branch fails the linting test intentionally on Circle to remind me to remove the changes to `MyDomains` and unmount the middleware before merging.

**Testing regular accounts**
- Restart your server after checking out the branch
- Log out of WordPress.com
- Log in to Delphin with an account without 2FA enabled
- Check that you are logged in to WordPress.com
- Log out of WordPress.com again
- Click on `My Domains` in the footer, then on Learn More, then on `My Domains` again. (this is a bug)
- Check that a new window opens and you are logged in to WordPress.com

**Testing 2FA accounts**
- Restart your server after checking out the branch
- Log out of WordPress.com
- Log in to Delphin with an account with 2FA enabled
- Check that you are not logged in to WordPress.com
- Click on `My Domains` in the footer, then on Learn More, then on `My Domains` again. (this is a bug)
- Check that a new window opens and you are are prompted to enter your 2FA code to log in to WordPress.com

I am dispatching the log in event on this page to make this middleware possible to test, but that will be reverted before merging.
